### PR TITLE
Use React

### PR DESCRIPTION
**Pull request by ptbrowne**
Use react@16.2 if `process.env.USE_REACT` is true during build.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -3,5 +3,10 @@
   "extends": ["cozy-app/react"],
   "globals": {
     "jest": true
+  },
+  "settings": {
+    "react": {
+      "version": "15.6.2", // React version, default to the latest React stable release
```
Comment (drazik) [inline .eslintrc]: Why not `16.5.2` ? Actually it seems you inverted the `5` and the `6`.

Comment (ptbrowne) [reply .eslintrc]: Thanks, it is 16.2, I had it before upgrading to 16, and cozy-ui actually needs the 16 for the Portal to works. 16.2 and not 16.3 not to have the lifecycle hooks deprecations (componentDidUpdate -> UNSAFE_componentDidUpdate in 16.3)

File: src/ducks/settings/TogglePane.jsx [hunk 2]
```diff
@@ -82,9 +81,4 @@ TogglePane.propTypes = {
   saveDocument: PropTypes.func.isRequired
 }
 
-export default compose(
-  translate(),
-  // We keep `connect` just so the component still receive the `dispatch` function as prop
-  // This has to be removed when we handle the mutations with the new cozy-client
-  connect()
-)(TogglePane)
+export default compose(translate())(TogglePane)
```
Comment (kosssi) [inline src/ducks/settings/TogglePane.jsx]: Il y a plus besoin de compose du coup.

```suggestion
export default translate()(TogglePane)
```

File: package.json [hunk 3]
```diff
@@ -122,11 +122,11 @@
     "postcss-loader": "3.0.0",
     "pretty": "2.0.0",
     "prop-types": "15.6.2",
-    "react": "15.6.2",
-    "react-addons-test-utils": "15.6.2",
-    "react-dom": "15.6.2",
+    "react": "16.2",
+    "react-addons-test-utils": "^16.0.0-alpha.3",
```
Comment (kosssi) [inline package.json]: renovate repassera par la puisqu'il fixe les versions donc sans `^`

File: .eslintrc (hunk 1 above)
Comment (drazik) [reply .eslintrc]: :+1:

**Pull request merged**